diff --git a/lib/index.js b/lib/index.js
index 5a75f390e1ed59db192587176cf6fabd229cdf84..f945ec700106c4b8ea31cd4f3377cf302017fa98 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -52,7 +52,8 @@ export function initAuth(config) {
         new Headers(request.headers), config).then(async (authResponse) => {
             const { user = null, expires = null, ...rest } = (await authResponse.json()) ?? {};
             // Preserve cookies set by Auth.js Core
-            const cookies = authResponse.headers.get("set-cookie");
+            const cookies = authResponse.headers.getSetCookie();
+            console.log("next-auth, lib/index 1", authCookies.length)
             if (cookies)
                 response?.setHeader("set-cookie", cookies);
             return { user, expires, ...rest };
@@ -100,7 +101,8 @@ async function handleAuth(args, config, userMiddlewareOrRoute) {
     }
     // Preserve cookies set by Auth.js Core
     const finalResponse = new Response(response?.body, response);
-    const authCookies = sessionResponse.headers.get("set-cookie");
+    const authCookies = sessionResponse.headers.getSetCookie();
+    console.log("next-auth, lib/index 2", authCookies.length)
     if (authCookies)
         finalResponse.headers.set("set-cookie", authCookies);
     return finalResponse;
